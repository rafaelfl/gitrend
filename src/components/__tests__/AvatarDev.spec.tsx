import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { AvatarDev } from '..';
import testProfile from '../../assets/images/test-profile.jpg';
import octocatAvatar from '../../assets/images/octocat.svg';
import userEvent from '@testing-library/user-event';

describe('AvatarDev component tests', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    test('it should match snapshot', () => {
        const { container } = render(<AvatarDev />);

        expect(container).toMatchSnapshot();
    });

    test('it should render an image as the dev test avatar', () => {
        const { getByRole, container } = render(<AvatarDev src={testProfile} />);

        const avatarContainer = container.querySelector('.avatar-container');
        const avatar = getByRole('button');

        expect(avatarContainer).toBeInTheDocument();
        expect(avatar).toBeInTheDocument();

        expect(avatar).toHaveStyle(`background-image: url(${testProfile})`);
        expect(avatar).not.toHaveStyle(`background-image: url(${octocatAvatar})`);
    });

    test('it should render an image as octocat', () => {
        const { getByRole, container } = render(<AvatarDev />);

        const avatarContainer = container.querySelector('.avatar-container');
        const avatar = getByRole('button');

        expect(avatarContainer).toBeInTheDocument();
        expect(avatar).toBeInTheDocument();

        expect(avatar).not.toHaveStyle(`background-image: url(${testProfile})`);
        expect(avatar).toHaveStyle(`background-image: url(${octocatAvatar})`);
    });

    test('it should call the onClick event', () => {
        const onClick = jest.fn();

        const { getByRole, container } = render(<AvatarDev onClick={onClick} />);

        const avatarContainer = container.querySelector('.avatar-container');
        const avatar = getByRole('button');

        expect(avatarContainer).toBeInTheDocument();
        expect(avatar).toBeInTheDocument();

        userEvent.click(avatar);

        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
