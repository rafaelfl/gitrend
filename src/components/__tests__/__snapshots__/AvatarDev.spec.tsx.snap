// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarDev component tests it should match snapshot 1`] = `
<div>
  <div
    class="avatar-container"
  >
    <button
      class="avatar-container__button"
      style="background-image: url(octocat.svg);"
    />
  </div>
</div>
`;
